### PR TITLE
Issue/2440

### LINKED ISSUE #2440: Clicking on Save Changes button under Settings should not trigger popup
## Issue Overview
While changing settings, I've noticed that on clicking the `Save Changes` button, the confirmation popup for leaving the page on setting change appears which should not appear on clicking `Save Changes` button as i am about to save those changes.

Check GIF for more details:
gif

## Expected Behavior
On clicking `Save Changes` button after changing settings should directly save the changed settings.

## Current Behavior
On clicking `Save Changes` button after changing settings display popup to confirm.

## Todos
- [ ] Must check all the possible areas and browsers
- [ ] Investigate Cause of Issue
- [ ] Fix Issue
- [ ] Test
- [ ] Create PR

## WordPress Environment
<details>

### WordPress Environment ###

Home URL: http://givetest.local
Site URL: http://givetest.local
WP Version: 4.9.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
Permalink Structure: /%postname%/
Show on Front: posts
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Inaccessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: givetest.local
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: nginx/1.10.1
PHP Version: ❌ 5.3.29 - We recommend a minimum PHP version of 5.6. See: PHP Requirements in Give
PHP Post Max Size: 1,000 MB
PHP Time Limit: 900
PHP Max Input Vars: 3000
PHP Max Upload Size: 1,000 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.2h - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: ❌ 5.5.55 - We recommend a minimum MySQL version of 5.6. See: WordPress Requirements
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 1.8.17
Upgraded From: –
Test Mode: Enabled
Currency Code: USD
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://givetest.local/donation-confirmation/
Failure Page: http://givetest.local/donation-failed/
Donation History Page: http://givetest.local/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation, PayPal Standard
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Admin Email Notifications: Enabled
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Other Active Plugins ###

Query Monitor: by John Blackbourn – 2.17.0

### Theme ###

Name: Twenty Seventeen
Version: 1.4
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

</details>

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR resolves #2440

## How Has This Been Tested?
I have tested this manually changing different types of settings and then leaving the page as well as saving the changes.

## Screenshots (jpeg or gifs if applicable):
gif

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: assets/js/admin/admin-scripts.js [hunk 1]
```diff
@@ -721,7 +672,63 @@ var give_setting_edit = false;
...
+			});
+		},
+
+		/**
+		 * Show alert when admin try to reload the page with saving the changes.
+		 *
+		 * @since 1.8.14
+		 */
+		changeSettingsUnload: function() {
+			if ( $( 'body' ).closest( '.post-type-give_forms' ).length > 0 ) {
```
Comment (ravinderk) [inline assets/js/admin/admin-scripts.js]: @mehul0810 Can we improve this selector

Comment (mehul0810) [reply assets/js/admin/admin-scripts.js]: @ravinderk I have improved the selector. Please check and let me know if there is any further change.

File: assets/js/admin/admin-scripts.js [hunk 2]
```diff
@@ -721,7 +672,63 @@ var give_setting_edit = false;
 					});
 				}
 			}).change();
+		},
+
+		saveButtonTriggered: function() {
+			$( 'body' ).on( 'click', '.give-save-button', function () {
```
Comment (ravinderk) [inline assets/js/admin/admin-scripts.js]: @mehul0810 update this selector also

Comment (mehul0810) [reply assets/js/admin/admin-scripts.js]: @ravinderk Updated this selector as well.

File: assets/js/admin/admin-scripts.js [hunk 3]
```diff
@@ -721,7 +672,63 @@ var give_setting_edit = false;
 					});
 				}
 			}).change();
+		},
+
+		saveButtonTriggered: function() {
+			$( '.give-save-button' ).on( 'click', function () {
```
Comment (ravinderk) [inline assets/js/admin/admin-scripts.js]: @mehul0810 This should be setting page specific. We can have `.give-save-button` at multiple place.

Comment (mehul0810) [reply assets/js/admin/admin-scripts.js]: @ravinderk Updated the code to be more specific for settings page.

**Pull request merged**